linera-execution/src/system.rs

```diff
@@ -530,35 +530,33 @@ where
                     SystemExecutionError::AdminOperationOnNonAdminChain
                 );
                 match admin_operation {
-                    AdminOperation::CreateCommittee { epoch, committee } => {
+                    AdminOperation::CreateCommittee { epoch, blob_hash } => {
                         ensure!(
                             epoch == self.epoch.get().expect("chain is active").try_add_one()?,
                             SystemExecutionError::InvalidCommitteeCreation
                         );
-                        self.committees.get_mut().insert(epoch, committee.clone());
+                        let blob_id = BlobId::new(blob_hash, BlobType::Committee);
+                        let committee =
+                            bcs::from_bytes(self.read_blob_content(blob_id).await?.bytes())?;
+                        self.blob_used(Some(txn_tracker), blob_id).await?;
+                        self.committees.get_mut().insert(epoch, committee);
                         self.epoch.set(Some(epoch));
-                        let message = RawOutgoingMessage {
-                            destination: Destination::Subscribers(SystemChannel::Admin.name()),
-                            authenticated: false,
-                            grant: Amount::ZERO,
-                            kind: MessageKind::Protected,
-                            message: SystemMessage::CreateCommittee { epoch, committee },
-                        };
-                        outcome.messages.push(message);
+                        txn_tracker.add_event(
+                            StreamId::system(EPOCH_STREAM_NAME),
+                            bcs::to_bytes(&epoch)?,
+                            bcs::to_bytes(&blob_hash)?,
+                        );
                     }
                     AdminOperation::RemoveCommittee { epoch } => {
```

Comment:
It is a little strange to have such a formulation.
I would expect `RemoveCommittee { committee }`.
Then maybe rename `RemoveFromCommittee`.
Not blocker of course.

Reply:
I agree, it should probably be called `RevokeEpoch`? But then I'd rename the other one also from `CreateCommittee` to `CreateEpoch`, even though the latter contains a committee and the epoch is implied.

Not sure what the best names are.

Reply:
I created https://github.com/linera-io/linera-protocol/issues/3516 for this.